DataFormats/Math/interface/angle_units.h

```diff
@@ -0,0 +1,65 @@
+#ifndef DataFormats_Math_Angle_Units_h
+#define DataFormats_Math_Angle_Units_h
+
+#include <cmath>
+
+namespace angle_units {
+
+  constexpr long double piRadians(M_PIl);              // M_PIl is long double version of pi
+  constexpr long double degPerRad = 180. / piRadians;  // Degrees per radian
+
+  namespace operators {
+
+    // Angle
+    constexpr long double operator"" _pi(long double x) { return x * piRadians; }
+    constexpr long double operator"" _pi(unsigned long long int x) { return x * piRadians; }
+    constexpr long double operator"" _deg(long double deg) { return deg / degPerRad; }
+    constexpr long double operator"" _deg(unsigned long long int deg) { return deg / degPerRad; }
+    constexpr long double operator"" _rad(long double rad) { return rad * 1.; }
+
+    template <class NumType>
+    inline constexpr NumType convertRadToDeg(NumType radians)  // Radians -> degrees
+    {
+      return (radians * degPerRad);
+    }
+
+    template <class NumType>
+    inline constexpr long double convertDegToRad(NumType degrees)  // Degrees -> radians
+    {
+      return (degrees / degPerRad);
+    }
+  }  // namespace operators
+}  // namespace angle_units
+
+namespace angle0to2pi {
+
+  using angle_units::operators::operator""_pi;
+
+  // make0To2pi constrains an angle to be >= 0 and < 2pi.
+  // This function is a faster version of reco::reduceRange.
+  // In timing tests, it is almost always faster than reco::reduceRange.
+  // It also protects against floating-point value drift over repeated calculations.
+  // This implementation uses multiplication instead of division and avoids
+  // calling fmod to improve performance.
+
+  template <class valType>
+  inline constexpr valType make0To2pi(valType angle) {
```

Comment:
the most recent discussion with @cvuosalo seemed to have come to a conclusion that this method should stay in the deltaPhi.h file ( even with a pending renaming to the normal reco namespace and from make0To2pi to ```reduceRange0To2pi```

Reply:
Since this PR original purpose was targeted to facilitate the DD4hep migration in order to solve overlaps due to the units mismatch I'd suggest we decouple this PR from further API improvements in DataFormats that can be done separately.

Reply:
It sounds like this comment is in support of removing make0To2pi from this file because it is not relevant to the DD4hep migration

Reply:
IMHO, where it is now - e.g. in angle0to2pi namespace in angle_units.h and not in reco namespace in deltaPhi.h - is a good place since it's used only in a geometry Phi.h:
https://cmssdt.cern.ch/lxr/search?%21v=CMSSW_11_0_X_2019-07-31-2300&_filestring=&_string=make0To2pi

Reply:
OK, I'll wait for @cvuosalo to confirm, since yesterday I bought into the argument that deltaPhi is the right place for this make0To2pi.

Reply:
@slava77 - ok, done.